python/common/org/Python.java

```diff
@@ -394,10 +394,65 @@ public static void initializeModule(java.lang.Class cls, java.util.Map<java.lang
                     "As repr(), return a string containing a printable representation of an\n" +
                     "object, but escape the non-ASCII characters in the string returned by\n" +
                     "repr() using \\x, \\u or \\U escapes.  This generates a string similar\n" +
-                    "to that returned by repr() in Python 2.\n"
+                    "to that returned by repr() in Python 2.\n",
+            args = {"object"}
     )
-    public static org.python.types.Str ascii() {
-        throw new org.python.exceptions.NotImplementedError("Builtin function 'ascii' not implemented");
+    public static org.python.types.Str ascii(org.python.Object object) {
+        boolean isString = false;
+        if (object instanceof  org.python.types.Str) {
+            isString = true;
+        }
+        java.lang.String s = ((org.python.types.Str) object.__repr__()).value;
+        StringBuilder resultString = new StringBuilder();
+
+        if (s.length() == 3) {
+            if (s.charAt(0) == 39 && s.charAt(1) == 39 && s.charAt(2) == 39) {
```

Comment:
I see. If that's the case, there definitely should be a test for it.

What happens if the input is just one or two `'`? Or some other number of `'`?